src/components/Terminal/ProblemsTab.tsx

```diff
@@ -1,73 +1,299 @@
+      <span className="text-xs">Diagnostics</span>
+      <ul className='flex flex-row items-center'>
+        {errorDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1.5 text-xs bg-red-500'>
+              {errorDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {warningDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-yellow-500'>
+              {warningDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {infoDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {infoDiagnostics.length}
+            </span>
+          </li>
+        )}
+        {hintDiagnostics.length > 0 && (
+          <li>
+            <span className='inline-block rounded-full px-1.5 ml-1 text-xs bg-blue-500'>
+              {hintDiagnostics.length}
+            </span>
+          </li>
+        )}
+      </ul>
+    </div>
+  );
+};
+
+interface SeverityIconProps {
+  severity: Diagnostic['severity']
+}
+
+const SeverityIcon: React.FunctionComponent<SeverityIconProps> = ({ severity }) => {
+  switch (severity) {
+  case 1: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscWarning className='text-yellow-500 w-4 h-4' />
     </div>
   );
+  case 2: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscInfo className='text-blue-500 w-4 h-4' />
+    </div>
+  );
+  case 3: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscLightbulb className='text-green-500 w-4 h-4' />
+    </div>
+  );
+  default: return (
+    <div className='flex flex-row items-center justify-center'>
+      <VscError className='text-red-500 w-4 h-4' />
+    </div>
+  );
+  }
+};
+
+type ActiveSeverity = DiagnosticSeverity | 'all';
+
+interface SeverityButtonsProps {
+  diagnostics: Diagnostic[];
+  active: ActiveSeverity;
+  setActive: React.Dispatch<React.SetStateAction<ActiveSeverity>>;
+}
+
+const SeverityButtons: React.FunctionComponent<SeverityButtonsProps> = ({ diagnostics, active, setActive: defaultSetActive }) => {
+  console.log(diagnostics);
+
+  const errorDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 0);
+  const warningDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 1);
+  const infoDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 2);
+  const hintDiagnostics = diagnostics.filter(diagnostic => diagnostic.severity === 3);
+
+  const activeBg = 'bg-gray-900';
+  const notActiveBg = 'bg-gray-700';
+
+  const setActive = useCallback((type: ActiveSeverity) => {
+    defaultSetActive(type === active ? 'all' : type);
+  }, [active]);
+
+  return (
+    <ul className='flex flex-row items-center'>
+      <Tooltip content="Show only errors">
```

Comment:
The tooltip is great if this option is not selected but when I click on the button it still says "Show only errors", and it's wrong. It should probably say "Show everything" or "Stop showing only errors" or something like that.

That said, playing with these buttons I realized it's acting as a "select" or "option". Wouldn't it make more sense to accept selecting multiple options? For instance, I want to see errors and warnings only. This is a common thing when you're fixing problems and don't want to have much noise. Also, tooltips will become easier. They'd be something like "Show errors" or "Hide errors".

Reply:
Ok, I implemented this second approach with `Show all diagnostics` text if user select only one option :) Play with it and you will see what I have in mind. Thanks for suggestion!

Reply:
![Captura de Pantalla 2022-11-15 a las 18 35 15](https://user-images.githubusercontent.com/242119/201987835-5af2dcb3-a6cf-4d3e-bbc2-e7b06c7870b9.png)

The default state doesn't make sense. It says "Show warnings" but I'm already seeing warnings 😅 I think, by default, all buttons should be checked.

Reply:
> I think, by default, all buttons should be checked.

It doesn't have a sense, because they people will need to uncheck three buttons if they will want to see only warnings. I changed state to show messages like:
- any button is checked: `Show only {diagnostic type}`
- one button is checked and we hover this checked button: `Show all diagnostics` and for other `Show {diagnostic type}`
- for rest: `Show {diagnostic type}` or `Hide {diagnostic type}`

WDYT? :) You can check in preview :)

Reply:
Ok. Fair. It's weird that all buttons checked and all buttons unchecked will show you the same thing 😅

Reply:
> Ok. Fair. It's weird that all buttons checked and all buttons unchecked will show you the same thing 😅

We should leave it as it's now, because people should have option to disable only one particular kind of diagnostics, when you have enabled others. The current solution isn't perfect but by changing it, people may have a worse UX.

> And it lets me disable warning, information, and hint messages. In such case, I think we should disable the buttons in this menu too and when you hover over them you'll see a tooltip saying it's been disabled in the settings and that you have to enable them back if you want to use them.

I will add.

Reply:
![image](https://user-images.githubusercontent.com/20404945/203025059-697972a3-0e1b-4c54-8fd4-1da5d2a2f955.png)

Can it be?

Reply:
Alright, this is looking much better now 👏